src/System.Runtime.Extensions/tests/System/EnvironmentTests.cs

```diff
@@ -191,6 +191,55 @@ public void FailFast_ExpectFailureExitCode()
             }
         }
 
+        [Trait(XunitConstants.Category, XunitConstants.IgnoreForCI)] // fail fast crashes the process
+        [Fact]
+        public void FailFast_InnerExceptionStackTrace()
+        {
+            var psi = new ProcessStartInfo();
+            psi.RedirectStandardError = true;
+
+            using (RemoteInvokeHandle handle = RemoteInvoke(
+                () => { Environment.FailFast("message", new ArgumentException("bad arg")); return SuccessExitCode; },
+                new RemoteInvokeOptions { StartInfo = psi }))
+            {
+                Process p = handle.Process;
+                handle.Process = null;
+                p.WaitForExit();
+                String consoleOutput = p.StandardError.ReadToEnd();
+                Assert.True(consoleOutput.Contains("Exception details:"));
+                Assert.True(consoleOutput.Contains("ArgumentException"));
+                Assert.True(consoleOutput.Contains("bad arg"));
+            }
+            // Test using another type of exception
```

Comment:
This method contains 3 independent tests, I would put them in separate methods. It is good to have short tests if possible as it makes it easier to read and quicker to run down failures.